### ISSUE TITLE
Add reticulating splines

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#18228]

**Comment by ara4n** [comment]
This could also help a lot with the leftpanel being blank (why? surely the client has valid contents for the leftpanel at this point...):

<img width="1453" alt="Screenshot 2021-07-26 at 15 12 26" src="https://user-images.githubusercontent.com/1294269/127003424-b17e11fa-033c-48d1-b183-e98383b03c8b.png">

**Comment by t3chguy** [comment]
iirc the room list waits for the whole sync to be processed before showing anything so that it doesn't move around/message previews update until a consistent state

**Comment by ara4n** [comment]
this doesn't feel right. surely we shouldn't stop people reading their history during catchup. we could freeze the contents of the list or show greeked-out previews until it's caught up, rather than hide it entirely?  Otherwise this undermines all the great work @gsouquet did to speed up the ability to the app on launch (if the UI to change room is then hidden from you, despite the fact that changing room work correctly and triggers a prioritised catchup).

**Comment by turt2live** [comment]
The room list is hidden because it will melt your battery and be generally unresponsive until it is loaded. The multi-dispatcher work will likely help with the responsiveness, but we don't have a stable point in time to run the complex room list calculations until later in the app load.

**Comment by ara4n** [comment]
yup, we could just show stale roomlist until it's synced, which would be better than zero roomlist.

**Comment by turt2live** [comment]
it'd be a non-functional stale room list: clicking would be lagged by several seconds.

**Comment by ara4n** [comment]
i'm proposing you could stop the roomlist from being updated until the sync completes - but at least start off showing stale content so you can switch room.

https://twitter.com/netzverweigerer/status/1415703860845420545 for more snark pointed at this.

**Comment by turt2live** [comment]
right, but the dispatcher is so busy during that time that it'd fail to work properly, even for simple room switches. Plus the intense amount of CPU used to re-render the thing when the sync does complete eventually. It not showing is us avoiding a lot of the React-level performance issues that the room list encounters. 

These are all fixable things, but we'd probably want to prioritize the redesign of it to avoid having to do all the work twice (again).

**Comment by JeanPaulLucien** [comment]
I have seen a similar screenshot in #17141
